execution/src/worker.rs

```diff
@@ -0,0 +1,171 @@
+use crate::config::ExecutionConfig;
+use crate::error::ExecutionError;
+use models::{address::AddressHashMap, Amount, Block, BlockHashMap, BlockId, Slot};
+use tokio::sync::mpsc;
+
+/// Commands sent to the `execution` component.
+pub enum ExecutionCommand {
+    /// The clique has changed,
+    /// contains the blocks of the new blockclique
+    /// and a list of blocks that became final
+    BlockCliqueChanged {
+        blockclique: BlockHashMap<Block>,
+        finalized_blocks: BlockHashMap<Block>,
+    },
+}
+
+// Events produced by the execution component.
+pub enum ExecutionEvent {
+    /// A coin transfer
+    /// from the SCE ledger to the CSS ledger.
+    TransferToConsensus,
+}
+
+/// Management commands sent to the `execution` component.
+pub enum ExecutionManagementCommand {}
+
+struct SCELedgerEntry {
+    balance: Amount,
+    bytecode: Vec<u8>,
+    data: Vec<u8>,
+}
+
+pub struct ExecutionWorker {
+    /// Configuration
+    _cfg: ExecutionConfig,
+    /// Thread count
+    thread_count: u8,
+    /// Receiver of commands.
+    controller_command_rx: mpsc::Receiver<ExecutionCommand>,
+    /// Receiver of management commands.
+    controller_manager_rx: mpsc::Receiver<ExecutionManagementCommand>,
+    /// Sender of events.
+    _event_sender: mpsc::UnboundedSender<ExecutionEvent>,
+    /// Time cursors
+    last_final_slot: Slot,
+    last_active_slot: Slot,
+    /// SCE ledger
+    final_ledger: AddressHashMap<SCELedgerEntry>,
+    active_ledger: AddressHashMap<SCELedgerEntry>,
+    /// ordered active blocks
```

Comment:
Does that mean blocks that are not yet SCE final that were processed by the SCE ?

Reply:
some blocks are CSS-final but not SCE-final. "ordered active blocks" includes all non-SCE-final blocks

Reply:
Whether there were or not processed by the SCE ?

Reply:
check the source code, for now blocks are processed as they SCE-finalize to update the final_ledger